HotelBase/HotelBaseClass.cs

```diff
@@ -7,51 +7,51 @@ public abstract class HotelBaseClass
     {
         private readonly IBookingService _bookingService;
         private readonly IPaymentService _paymentService;
+        internal readonly ILogger Logger;
 
         internal bool MakePayment(int creditCardNumber, double price)
         {
-            bool success;
+            bool successfulPayment;
             try
             {
-                success = _paymentService.Pay(creditCardNumber, price);
+                successfulPayment = _paymentService.Pay(creditCardNumber, price);
             }
             catch (Exception e)
             {
-                Console.WriteLine(e.Message);
-                return false;
+                Logger.Write(e.Message);
```

Comment:
Nice to have:

I am not sure what you try to do by catching exception log just exception message and rethrowing it. Unless I I have missed somthing you can do one of:
* Add some info to logger, which allow to identify payment error
* Keep generic exception message, but then you doesn't need to catch it here, you can do that much higher e.g. in Program file.